### PR TITLE
Feature/add ocaml kernel

### LINKED ISSUE #333: [Feature]: Add back the OCaml Kernel
### What's missing?

The new version (Flake rewrite) does not feature anymore the OCaml Kernel, that I added (#162).
I would like to continue using this project with the OCaml kernel for jupyter.

### Possible solution

It should only be a rewrite of the old merged solution. I will try myself to make it work and submit a PR by the end of the week.

### Alternatives

No particular alternative.

### Additional context

Feature was previously implemented by GH-163, closing GH-162.

### PR DESCRIPTION
**Pull request by djacu**
Fixes #333

### CODE REVIEW

**Comment by fusetim** [comment]
Wow you are fast! 
It works great, and I don't think we can do better there. I tried to use findlib (the package that should be used in this type of situation) but did not seem to help.

I note that you hardcode the ocaml version in the `CAML_LD_LIBRARY_PATH` when `pkgs` and `ocamlPackages` can be edit to use another version of Ocaml.

Great job!

**Comment by djacu** [comment]
Well you did most of the work. I just brought it over to the new architecture. ;)

Oh good catch! I just submitted another commit so it will dynamically pull the correct version if a user overrides `ocamlPackages`.

**Comment by fusetim** [comment]
Seems good to me :) Perfectly working on my side.

**Comment by djacu** [comment]
Just waiting for some time with one of the codeowners. We'll merge next time we get a chance to meet.

File: kernels/available/ocaml/default.nix [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  wrappedEnv = let
+    ocamlVersion = ocamlPackages.ocaml.version;
+  in
+    pkgs.runCommand "wrapper-${env.name}"
+    {nativeBuildInputs = [pkgs.makeWrapper];}
+    ''
+      mkdir -p $out/bin
+      for i in ${env}/bin/*; do
+        filename=$(basename $i)
+        makeWrapper ${env}/bin/$filename $out/bin/$filename \
```
Comment (thufschmitt) [inline kernels/available/ocaml/default.nix:76]:
```suggestion
        # XXX: 'CAML_LD_LIBRARY_PATH' should be set automatically by the kernel.
        # Not sure why it doesn't, but setting it manually seems to fix things
        makeWrapper ${env}/bin/$filename $out/bin/$filename \
```

**Review by thufschmitt — APPROVED** [review]

**Comment by djacu** [comment]
> Just noticed that this doesn't include the OCaml logo to the list in the readme. Should it be added?

Oh that's my mistake. I'll add it in another PR.

**Pull request merged**